BrainPortal/app/models/cluster_task.rb

```diff
@@ -2350,6 +2330,46 @@ def validate_json_string json_string
     JSON::Validator.validate!(schema,json_string) # raises an exception if json_string is not valid
   end
 
+  ##################################################################
+  # Singularity load
+  ##################################################################
+
+  private
+
+  def container_image_name #:nodoc:
+    ".container-#{self.run_id}.img"
+  end
+
+  # Install the singularity image as a local file.
+  def load_singularity_image #:nodoc:
+    singularity_image_name = self.tool_config.containerhub_image_name
+
+    # Singularity PULL
```

Comment:
1) I would give priority to the singularity local image over the singularity pull
2) I would split this method into three:
   one method does the if/else
   one method for sing pull
   on method for local sing image
3) I woudl add a addlog() before doing singularity pull so that the task's log